AU Core MedicationStatement addition

dt-r opened the pull request:
Addresses Jira: [FHIR-49960](https://jira.hl7.org/browse/FHIR-49960), [FHIR-49954](https://jira.hl7.org/browse/FHIR-49954), [FHIR-49956](https://jira.hl7.org/browse/FHIR-49956), [FHIR-49957](https://jira.hl7.org/browse/FHIR-49957), [FHIR-49958](https://jira.hl7.org/browse/FHIR-49958), [FHIR-49959](https://jira.hl7.org/browse/FHIR-49959)

Adding AU Core MedicationStatement. Changes to:
* general conformance
* add au core medication statement
* add to au core responder & requester capability statements incl bringing in DiagnosticReport
* change log
* comparison
* examples
* aucdi mapping

JayMurdoch commented on input/resources/au-core-medicationstatement.xml:
```diff
@@ -0,0 +1,264 @@
+        <rules value="open"/>
+      </slicing>
+    </element>
+    <element id="MedicationStatement.medication[x]:medicationCodeableConcept.coding:pbs">
```
Confirming if [PBS Item Codes](https://build.fhir.org/ig/hl7au/au-fhir-base/ValueSet-pbs-item.html) should be bound here?

dt-r replied:
@JayMurdoch : is the review comment stating that there is a missing slice for PBS Item Codes?

JayMurdoch replied:
Apologies - the review comment was not clear. Existing slice was missing binding to PBS Item Codes. Confirmed as fixed.

JayMurdoch commented on input/resources/au-core-medicationstatement.xml:
```diff
@@ -0,0 +1,264 @@
+      <sliceName value="pbs"/>
+      <mustSupport value="true"/>
+    </element>
+    <element id="MedicationStatement.medication[x]:medicationCodeableConcept.coding:amt">
```
Confirming if [Australian Medication](https://tx.ontoserver.csiro.au/fhir/ValueSet/australian-medication-1) should be bound here?

dt-r replied:
@JayMurdoch : is the review comment stating that there is a missing slice for Australian Medication or that there is something technically incorrect with the existing slice?

JayMurdoch replied:
Apologies - the review comment was not clear. Existing slice was missing binding to Australian Medication. Confirmed as fixed.

JayMurdoch commented on input/intro-notes/StructureDefinition-au-core-medicationstatement-notes.md:
```diff
@@ -0,0 +1,104 @@
+        <td>A requester <b>SHALL</b> provide a value precise to the second + time offset. A responder <b>SHALL</b> support a value precise to the second + time offset.<br/><br/>The requester <strong>SHALL</strong> support these search comparators <code>gt</code>, <code>lt</code>, <code>ge</code>, <code>le</code>. The responder <strong>SHALL</strong> support these search comparators <code>gt</code ...
+  </tr>
+  <tr>
+        <td>status</td>
```
Requester and Responder CapabilityStatements contain additional information not included in notes "The requester SHALL support multipleOr. 
 The responder SHALL support multipleOr."

dt-r replied:
@JayMurdoch: so, since the CapabilityStatements are normative. There is a review comment on the notes section that it is missing content?

dt-r replied:
Notes file for status are updated.

JayMurdoch replied:
Confirmed.

JayMurdoch commented on input/intro-notes/StructureDefinition-au-core-medicationstatement-notes.md:
```diff
@@ -0,0 +1,104 @@
+    - **SHOULD** support these **[`_include`](http://hl7.org/fhir/R4/search.html#include)** parameters: `MedicationStatement:medication`
+    - **SHOULD** support chained searching of patient canonical identifier `patient.identifier` (e.g. `patient.identifier=[system|][code]`)
+
+    `GET [base]/MedicationStatement?patient={Type/}[id]` or optionally`GET [base]/MedicationStatement?patient.identifier=[system|][code]`
```
Add space after 'or optionally'

dt-r replied:
fixed

JayMurdoch replied:
Confirmed.

JayMurdoch commented on input/intro-notes/StructureDefinition-au-core-medicationstatement-notes.md:
```diff
@@ -0,0 +1,104 @@
+    - **SHALL** support *[multipleOr](http://hl7.org/fhir/R4/searchparameter-definitions.html#SearchParameter.multipleOr)* search on `status` (e.g.`status={system|}[code],{system|}[code],...`)
+
+
+    `GET [base]/MedicationStatement?patient={Type/}[id]&status={system|}[code]{,{system|}[code],...}`
```
Remove bolded curly brackets: GET [base]/MedicationStatement?patient={Type/}[id]&status={system|}[code]**{**,{system|}[code],...**}**

dt-r replied:
The {,{.... is how optional multipleOr is shown. No change.

JayMurdoch replied:
Noted, thanks.

JayMurdoch commented on input/intro-notes/StructureDefinition-au-core-medicationstatement-notes.md:
```diff
@@ -0,0 +1,104 @@
+    *Implementation Notes:* Fetches a bundle of all MedicationStatement resources for the specified patient and status ([how to search by reference](http://hl7.org/fhir/R4/search.html#reference) and [how to search by token](http://hl7.org/fhir/R4/search.html#token))
+
+
+#### Optional Search Parameters
```
Example not provided for _id

dt-r replied:
Content for _id provided under `#### Optional Search Parameters`

JayMurdoch replied:
Confirmed.

JayMurdoch commented on input/intro-notes/StructureDefinition-au-core-medicationstatement-notes.md:
```diff
@@ -0,0 +1,104 @@
+    - **SHOULD** support these **[`_include`](http://hl7.org/fhir/R4/search.html#include)** parameters: `MedicationStatement:medication`
+    - **SHOULD** support chained searching of patient canonical identifier `patient.identifier` (e.g. `patient.identifier=[system|][code]`)
+    - **SHALL** support these `effective` comparators: `gt,lt,ge,le`
+    - **SHOULD** support *[multipleAnd](http://hl7.org/fhir/R4/searchparameter-definitions.html#SearchParameter.multipleAnd)* search on `effective` (e.g.`effective=[date]&effective=[date]]&...`)
```
Remove extra square bracket (bolded): (e.g.`effective=[date]&effective=[date]**]**&...`)

dt-r replied:
fixed

JayMurdoch replied:
Confirmed.

JayMurdoch commented on input/pagecontent/aucdi.md:
```diff
@@ -167,50 +167,49 @@ Column attribute descriptions are as follows:
     <td rowspan="9">Medication use statement</td>
     <td colspan="2">Medication name</td>
-    <td>AU Core MedicationStatement | <a href="StructureDefinition-au-core-medication.html">AU Core Medication</a></td>
+    <td><a href="StructureDefinition-au-core-medication.html">AU Core MedicationStatement</a> | <a href="StructureDefinition-au-core-medication.html">AU Core Medication</a></td>
```
AU Core MedicationStatement links to AU Core Medication

dt-r replied:
Fixed all instances in aucdi.

JayMurdoch replied:
Confirmed.

JayMurdoch commented on input/pagecontent/aucdi.md:
```diff
@@ -167,50 +167,49 @@ Column attribute descriptions are as follows:
   <tr>
     <td colspan="2">Clinical indication</td>
-    <td>AU Core MedicationStatement</td>
+    <td><a href="StructureDefinition-au-core-medication.html">AU Core MedicationStatement</a></td>
```
AU Core MedicationStatement links to AU Core Medication

JayMurdoch replied:
Change confirmed.

JayMurdoch commented on input/pagecontent/aucdi.md:
```diff
@@ -167,50 +167,49 @@ Column attribute descriptions are as follows:
   <tr>
     <td colspan="2">Strength</td>
-    <td>AU Core MedicationStatement | <a href="StructureDefinition-au-core-medication.html">AU Core Medication</a></td>
+    <td><a href="StructureDefinition-au-core-medication.html">AU Core MedicationStatement</a> | <a href="StructureDefinition-au-core-medication.html">AU Core Medication</a></td>
```
AU Core MedicationStatement links to AU Core Medication

JayMurdoch replied:
All entries confirmed.

JayMurdoch commented on input/pagecontent/comparison.md:
```diff
@@ -230,6 +236,20 @@ The following IPA profile(s) contain additional requirements. Implementers are a
+        </tr>
+         <tr>
+            <td rowspan="3" style="width: 25%;"><a href="StructureDefinition-au-core-medicationstatement.html">AU Core MedicationStatement</a></td>
+            <td rowspan="3" style="width: 25%;"><a href="https://hl7.org/fhir/uv/ipa/STU1/StructureDefinition-ipa-medicationstatement.html">IPA-MedicationStatement</a></td>
```
MedicationStatement.statusReason is MS element in IPA.

dt-r replied:
It is not MS in the referenced version of IPA which is 1.0.0. No change.

JayMurdoch replied:
I see statusReason as MS in 1.0.0 - please confirm.

dt-r replied:
Agree it's there, I checked the wrong profile. Now fixed.

JayMurdoch replied:
Confirmed.

JayMurdoch commented on input/pagecontent/comparison.md:
```diff
@@ -347,6 +366,24 @@ The following IPS profile(s) contain additional requirements. Implementers are a
+            <td style="width: 25%;"><i>Must Support</i> element in IPS.</td>
+        </tr>
+        <tr>   
+            <td style="width: 25%;">MedicationStatement.dosageInstruction.text</td>
```
Change element name to 'MedicationStatement.dosage.text'

dt-r replied:
fixed

JayMurdoch replied:
Confirmed.

JayMurdoch commented on input/pagecontent/comparison.md:
```diff
@@ -347,6 +366,24 @@ The following IPS profile(s) contain additional requirements. Implementers are a
+            <td style="width: 25%;"><i>Must Support</i> element in IPS.</td>
+        </tr>
+        <tr>   
+            <td style="width: 25%;">MedicationStatement.dosageInstruction.timing</td>
```
Change element name to 'MedicationStatement.dosage.timing'

dt-r replied:
fixed

JayMurdoch replied:
Confirmed.

JayMurdoch reviewed (changes requested):
See comments.

dt-r commented on input/pagecontent/comparison.md:
```diff
@@ -114,6 +114,12 @@ The table below provides a profile only comparison from AU Core to profiles in k
             <td style="width: 25%; text-align: center; vertical-align: middle;"><img src="orange_checkmark.svg.png" width="20"/></td>
         </tr>
+        <tr>
+            <td style="width: 25%; text-align: left; vertical-align: middle;"><a href="StructureDefinition-au-core-medicationrequest.html">AU Core MedicationStatement</a></td>
```
fixed.

JayMurdoch replied:
Confirmed.

dt-r commented:
@JayMurdoch, addressed review fixes + added examples.

dt-r commented:
@mjosborne1 please review terminology.

JayMurdoch commented on input/pagecontent/terminology.md:
```diff
@@ -136,6 +138,7 @@ Column attribute descriptions are as follows:
 |[LOINC](https://www.healthterminologies.gov.au/access-clinical-terminology/access-fhir-terminology-resources/code-systems/?ui:filter=loinc)|[Australian Pronouns](https://healthterminologies.gov.au/fhir/ValueSet/australian-pronouns-1), [RCPA SPIA Pathology Reporting](https://healthterminologies.gov.au/fhir/ValueSet/spia-pathology-reporting-1), [Vital Signs](http://hl7.org/fhir/R4/valueset-observa ...
 |[Medication request intent](https://hl7.org/fhir/R4/codesystem-medicationrequest-intent.html)|[Medication request intent](https://hl7.org/fhir/R4/valueset-medicationrequest-intent.html)|FHIR|Health Level Seven International|
 |[MedicationRequest status](https://hl7.org/fhir/R4/codesystem-medicationrequest-status.html)|[Medicationrequest status](https://hl7.org/fhir/R4/valueset-medicationrequest-status.html)|FHIR|Health Level Seven International|
+|[Medication status codes](https://hl7.org/fhir/R4/codesystem-medication-statement-status.html)|[Medication status codes](https://hl7.org/fhir/R4/valueset-medication-statement-status.html)|FHIR|
```
Confirmed.

mjosborne1 commented on input/examples/medicationstatement-active-bisoprolol.xml:
```diff
@@ -0,0 +1,44 @@
+                <coding>
+                    <system value="http://snomed.info/sct"/>
+                    <code value="23281011000036106"/>
+                    <display value="Bisoprolol fumarate 2.5 mg tablet"/>
```
While it's not incorrect to choose concepts for the same drug at different granularity as is the case here, perhaps an explanation as to why this example uses an AMT MPUU / Clinical drug and a PBS MPP in the same CodeableConcept might be of use, 1...so that other reviewers don't also call this out and 2...because it's a reasonable example of what may occur in the wild.

dt-r replied:
Noted comment but to make it easier to maintain the IG this 2nd coding is removed from this example.

mjosborne1 commented on input/examples/medicationstatement-active-bisoprolol.xml:
```diff
@@ -0,0 +1,44 @@
+            <manufacturer>
+                <identifier>
+                    <system value="http://pbs.gov.au/code/manufacturer"/>
+                    <value value="LR"/>
```
There is no Manufacturer in the PBS with a code = 'LR'.... Maybe due to legal issues singling out a single manufacturer? Perhaps need to call out this is deliberate.

dt-r replied:
removed from example as not marked MS or an AUCDI element.

mjosborne1 commented on input/examples/medicationstatement-active-bisoprolol.xml:
```diff
@@ -0,0 +1,44 @@
+    </contained>
+    <status value="active"/>
+    <medicationReference>
+        <reference value="#bisoprolol"/>
```
The reference to '#bisoprolol' flagged as missing in the QA report.

JayMurdoch replied:
FYI of note from testing - this QA error does not occur when only one coding is supplied for Medication.code.

dt-r replied:
Yes, this QA error is because we use PBS coding :(. Due to this, I will change which MedicationStatement example has a 2nd coding of PBS and instead put PBS coding on MedicationStatement-completed-chloramphenicol

mjosborne1 reviewed (approved):
See my comments inline.

JayMurdoch reviewed (changes requested):
Fixes and examples reviewed - one comment on the bisoprolol example QA error.

dt-r commented:
Please re-review. The following changes have been made to examples:
1. Changed the resource that has the 2nd pbs coding in it to avoid the QA error
2. Removed instances of MedicationStatement.context (neither MS nor AUCDI element)
3. Added a new MedicationStatement example to demonstrate MedicationStatement.reasonReference

JayMurdoch reviewed (changes requested):
One minor comment.

dt-r commented on input/examples/medicationstatement-completed-chloramphenicol.xml:
```diff
@@ -0,0 +1,30 @@
+        </coding>
+        <coding>
+            <system value="http://pbs.gov.au/code/item"/>
+            <code value="11112W"/>
```
Made suggested change.

JayMurdoch replied:
Confirmed.

steveswinsburg commented on input/ig.xml:
```diff
@@ -390,6 +390,39 @@
+				<reference value="MedicationStatement/previous-use-diflucan"/>
+			</reference>
+			<name value="MedicationStatement - patient has taken Diflucan in the past but is not taking it any more"/>
+			<description value="Shows an example of the previous use of Diflucan 200 mg/100 mL injection and the patient is not currently on this medicine at the time of the Home Medicines Review, for the *AU Core MedicationStatement* profile. This example demonstrates representation of additional AUCDI concepts for dosage including route, timing, dose and rate. Patient:  Mia Leanne Banks."/>
```
context removed from the example so maybe the narrative gets updated also to remove the review.

dt-r replied:
Fixed.
